From b7d78bde216a4ddaf528a849dbfb09e4c1e1a4a1 Mon Sep 17 00:00:00 2001
From: Orivej Desh <orivej@gmx.fr>
Date: Sat, 14 Sep 2024 09:48:36 +0000
Subject: [PATCH] Fix crash accessing uninitialized properties

When MuseScore is built with GCC 13.3.0 with optimizations, statements like

    return lineVisible() ? Sid::pedalEndText : Sid::pedalRosetteEndText;

in `Pedal::getPropertyStyle` are optimized to

    return Sid::pedalRosetteEndText - _lineVisible

However constructors like `Pedal::Pedal` call `EngravingObject::initElementStyle`
when such properties are not initialized yet, and when `initElementStyle` calls
`getPropertyStyle` it receives a bogus result and may crash when it calls
`setProperty` with a wrong `PropertyValue`.

On my machines this causes MuseScore to crash during startup about 40% of the
times. This change fixes the crash by adding default initialiazers.
---
 src/engraving/dom/property.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/engraving/dom/property.h b/src/engraving/dom/property.h
index 1ccf96cad2435..86422dd0d8b6d 100644
--- a/src/engraving/dom/property.h
+++ b/src/engraving/dom/property.h
@@ -44,7 +44,7 @@ namespace mu::engraving {
 //---------------------------------------------------------
 
 #define M_PROPERTY(a, b, c)                                      \
-    a _##b;                                                \
+    a _##b { };                                                \
 public:                                                     \
     const a& b() const { return _##b; }                  \
     void c(const a& val) { _##b = val; }                  \
